api/clients/v2/coretypes/derivation_errors.go

```diff
@@ -0,0 +1,98 @@
+package coretypes
+
+import (
+	"encoding/json"
+	"fmt"
+
+	_ "github.com/Layr-Labs/eigenda/api/clients/codecs"
+)
+
+// DerivationErrorStatusCode is an enum for the different error status codes
+// that can be returned during EigenDA "derivation" of a payload from a DA cert.
+// For more details, see the EigenDA spec on derivation:
+// https://github.com/Layr-Labs/eigenda/blob/f4ef5cd5/docs/spec/src/integration/spec/6-secure-integration.md#derivation-process
+//
+// This error is meant to be marshalled to JSON and returned as an HTTP 418 body
+// to indicate that the cert should be discarded from rollups' derivation pipeline.
```

Comment:
> rollups' derivation pipeline

grammar nit: "rollups`" is plural, but "pipeline" is singular (and different rollups don't share a single pipeline)

Options for these to agree:
1. "should be discarded from a rollup's derivation pipeline."
2. "should be discarded from rollups' derivation pipelines."